moveit_ros/warehouse/src/broadcast.cpp

```diff
@@ -73,7 +75,7 @@ int main(int argc, char** argv)
   node_options.allow_undeclared_parameters(true);
   node_options.automatically_declare_parameters_from_overrides(true);
   rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("publish_warehouse_data", node_options);
-  const auto logger = node->get_logger();
+  moveit::get_logger_mut() = node->get_logger();
```

Comment:
I'm not following this. 
`moveit::get_logger_mut()` returns this static logger right?
`static rclcpp::Logger logger = rclcpp::get_logger("moveit");`

How are we able to assign value?

Reply:
This works because get_logger_mut returns a reference to the static variable. Usually, it would be really bad to return a reference (or pointer) to a local variable, but in this case, it is static, which means it has a global lifetime. The first time the function is called, it gets set to the return of that get_logger function from ros, which is a logger that is not associated with a node; we can then assign to the static from the node logger.

I will write some tests today to verify this works as I think it does.

Reply:
What will happen if there are multiple nodes modifying the logger? Would there be a case where the wrong logger instance is used?

Reply:
If multiple nodes are created, and this function gets called multiple times, it will have the value of the second one. I don't know of any-place in moveit where that would happen, but it is one of the downsides to this approach.